_posts/2018-10-15-12-2-Use-SkipToken-for-Server-Side-Paging.md

```diff
@@ -0,0 +1,63 @@
+---
+layout: post
+title: "12.2 Use $skiptoken for server side paging "
+description: "WebAPI to use $skiptoken for server side paging"
+category: "12. Design"
+---
+# Use $skiptoken for server-driven paging
+
+### Background
+Loading large data can be slow. Services often rely on pagination to load the data incrementally to improve the response times and the user experience. Paging can be server-driven or client-driven:
+#### Client-driven paging
+In client-driven paging, the client decides how many records it wants to load and asks the server that many records. That is achieved by using $skip and $top tokens in conjunction. For instance, if a client needs to request 10 records from 71-80, it can send a similar request as below:
```

Comment:
$skip and $top tokens => tokens here is correct?

Reply:
Yes, it seems to be correct. I am talking about client driven paging here. Did I misunderstand your concern?

Reply:
We generally refer to these as "query options".  Tokens generally refer to server-encoded opaque values, like the skip token or delta token.

---
In reply to: [227582521](https://github.com/OData/WebApi/pull/1644#discussion_r227582521) [](ancestors = 227582521)